pandas/tests/indexes/test_multi.py

```diff
@@ -78,6 +78,49 @@ def test_labels_dtypes(self):
         self.assertTrue((i.labels[0] >= 0).all())
         self.assertTrue((i.labels[1] >= 0).all())
 
+    def test_repeat(self):
+        reps = 2
+        major_axis = ['foo', 'bar']
+        minor_axis = ['one', 'two']
+        major_labels = np.arange(7, 9)
+        minor_labels = np.arange(3, 5)
+        index_names = ['aaa', 'bbb']
+        m = MultiIndex(levels=[major_axis, minor_axis],
+                       labels=[major_labels, minor_labels],
+                       names=index_names, verify_integrity=False)
+        expected = MultiIndex(levels=[major_axis, minor_axis],
+                              labels=[major_labels.repeat(reps),
+                                      minor_labels.repeat(reps)],
+                              names=index_names, verify_integrity=False)
+        out = m.repeat(reps)
+
+        self.assertEqual(out.names, expected.names)
+        tm.assert_numpy_array_equal(out.levels, expected.levels)
+        tm.assert_numpy_array_equal(out.labels, expected.labels)
+
+    def test_numpy_repeat(self):
+        reps = 2
+        major_axis = ['foo', 'bar']
+        minor_axis = ['one', 'two']
+        major_labels = np.arange(7, 9)
+        minor_labels = np.arange(3, 5)
+        index_names = ['aaa', 'bbb']
+        m = MultiIndex(levels=[major_axis, minor_axis],
+                       labels=[major_labels, minor_labels],
+                       names=index_names, verify_integrity=False)
+        expected = MultiIndex(levels=[major_axis, minor_axis],
+                              labels=[major_labels.repeat(reps),
+                                      minor_labels.repeat(reps)],
+                              names=index_names, verify_integrity=False)
+        out = np.repeat(m, reps)
+
+        self.assertEqual(out.names, expected.names)
+        tm.assert_numpy_array_equal(out.levels, expected.levels)
```

Comment:
same here. btw, you can use ``pd.MultiIndex.from_product/array`` to easily create these types of things